test/e2e/ic_conditions_metric_test.go

```diff
@@ -0,0 +1,159 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	routev1client "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/openshift/library-go/test/library/metrics"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+// waitForIngressControllerConditionsMetrics waits for the metrics for ingress_controller_conditions to be present.
+func waitForIngressControllerConditionsMetrics(t *testing.T, prometheusClient prometheusv1.API) error {
+	t.Logf("Waiting for ingress_controller_conditions to be present")
+	if err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+		result, _, err := prometheusClient.Query(context, "ingress_controller_conditions", time.Now())
+		if err != nil {
+			t.Logf("Failed to fetch metrics: %v, retrying...", err)
+			return false, nil
+		}
+
+		// Check if fetched metrics is of Vector type.
+		vector, ok := result.(model.Vector)
+		if !ok {
+			t.Logf("Unexpected metric type, retrying...")
+			return false, nil
+		}
+
+		// Check if length of returned metric Vector is zero.
+		if len(vector) == 0 {
+			t.Logf("Metric is empty, retrying...")
+			return false, nil
+		}
+
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("Error waiting for ingress controller metrics: %w", err)
+	}
+	return nil
+}
+
+// restartOperatorPod will restart the operator pod, and if some error occurs it returns it
+func restartOperatorPod(t *testing.T, kubeClient kubernetes.Interface) error {
+	// Find the operator pod
+	t.Logf("Restarting Ingress operator pod...")
+	podsList, err := kubeClient.CoreV1().Pods("openshift-ingress-operator").List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	operatorPodName, operatorPodUID := extractIngressOperatorPodNameAndUID(podsList)
+
+	if operatorPodName == "" || operatorPodUID == "" {
+		return errors.New(("Unable to find ingress operator pod"))
+	}
+	// Delete the operator pod
+	if err := kubeClient.CoreV1().Pods("openshift-ingress-operator").Delete(context.Background(), operatorPodName, metav1.DeleteOptions{}); err != nil {
+		return err
+	}
```

Comment:
`List` or `Delete` can fail too (due to some networking or api server hiccups). I think it makes sense to wrap them into another `Poll` loop.